app/src/main/java/com/github/studydistractor/sdp/data/Event.kt

```diff
@@ -46,39 +50,42 @@ data class FirebaseEvent(
     val long: Double? = null,
     val start: String? = null,
     val end: String? = null,
-    val lateParticipation: Boolean = false,
+    val lateParticipation: Boolean? = null,
+    val numberOfPoints: Int? = null,
     val chatId: String? = null) {
     fun toEvent(): Event {
         return Event(
-            eventId, name!!, description!!, lat!!, long!!, start!!, end!!, lateParticipation, chatId!!
+            eventId!!,
+            name!!,
+            description!!,
+            lat!!,
+            long!!,
+            start!!,
+            end!!,
+            lateParticipation!!,
+            numberOfPoints!!,
+            chatId!!
         )
     }
 }
 
 /**
  * represent participants of a particular events
  *
- * @property eventId id of the event
  * @property participants list of userId that participate in this event
  */
 data class EventParticipants(
-    val eventId: String? = null,
-    val participants: List<String>? = null
+    val eventId: String = "",
+    val participants: Map<String, Boolean> = mapOf()
 )
 
-/**
- * Represent a chat that is received or sent from/to the firebase database
- *
- * @param chatId id of the chat
- * @param messageIds list of messageId that were sent in the chat
- */
 data class FirebaseEventParticipants(
-    val eventId: String?,
-    val participants: List<String>? = null
+    val eventId: String? = null,
+    val participants: Map<String, Boolean>? = null
 ) {
     fun toEventParticipants(): EventParticipants {
         return EventParticipants(
-            eventId, participants!!
+            eventId!!, participants!!

```

Comment:
Same comment as above

Reply:
I disagree on that one: Firebase dataclasses' whole point was to have nullable types that are casted to non-nullable types.